tests/helpers/oauth.ts

```diff
@@ -89,19 +99,25 @@ export async function createAccount(
 export type OAuthApplicationOptions = {
   scopes?: Schema.Scope[];
   redirectUris?: string[];
+  confidential?: boolean;
 };
 
 export async function createOAuthApplication(
-  options: OAuthApplicationOptions = { redirectUris: [OOB_REDIRECT_URI] },
+  options: OAuthApplicationOptions = {
+    redirectUris: [OOB_REDIRECT_URI],
+  },
 ): Promise<Pick<Schema.Application, "id">> {
   const clientId = base64.fromArrayBuffer(
     crypto.getRandomValues(new Uint8Array(16)).buffer as ArrayBuffer,
     true,
   );
-  const clientSecret = base64.fromArrayBuffer(
-    crypto.getRandomValues(new Uint8Array(32)).buffer as ArrayBuffer,
-    true,
-  );
+  const clientSecret =
+    options.confidential === true
+      ? base64.fromArrayBuffer(
+          crypto.getRandomValues(new Uint8Array(32)).buffer as ArrayBuffer,
+          true,
+        )
+      : "";
```

Comment:
This may be a little weird, but for now I don't want to chat the nullable constraint on `clientSecret` on the applications table, so instead a "public client" uses an empty string for the client secret, along with setting confidential to false.

Reply:
Because I was concerned about it being an unsafe migration: https://github.com/ankane/strong_migrations (see the readme, it has one that's on NOT NULLs)

Reply:
Yeah, I think so now, so we could make that change (though it might be better to leave as not nullable because otherwise any time we access the client_secret we'll need nullable checks which could get annoying